### ISSUE TITLE
@ContentChild() works fine with static true but according to the ts definitions it doesnt accept static true

### ISSUE THREAD

**Issue by bojidaryovchev** [angular/angular#36876]

**Comment by Achilles1515** [comment]
> ...I was wondering if its safe to rely on that as it is not part of the typescript definitions for the acceptable options
Where are you seeing this?

https://github.com/angular/angular/blob/master/packages/core/src/metadata/di.ts#L191

```ts
export interface ContentChildDecorator {
  /**
   * ...
   * **Metadata Properties**:
   *
   * * **selector** - The directive type or the name used for querying.
   * * **read** - True to read a different token from the queried element.
   * * **static** - True to resolve query results before change detection runs,
   * false to resolve after change detection. Defaults to false.
   * ...
   */
  (selector: Type<any>|Function|string, opts?: {read?: any, static?: boolean}): any;
  new(selector: Type<any>|Function|string, opts?: {read?: any, static?: boolean}): ContentChild;
}
```

![image](https://user-images.githubusercontent.com/23387905/80724430-d5ea8000-8acf-11ea-9082-d7f240e27ce5.png)

Are you think of `ContentChildren`?

**Comment by JoostK** [comment]
The `static` option has only been added in Angular 8.0 I believe, so if you're on an older version it won't be available.

**Comment by bojidaryovchev** [comment]
I am using Angular 9 and yes I meant @ContentChildren ty for pointing that out - because the @ContentChildren doesnt have an option for static true but works nevertheless

**Comment by JoostK** [comment]
I see. In that case, it is an implementation detail and it is **not** safe to rely on it. In fact, I would consider it a bug that the static flag of queries can be adjusted when they aren't supposed to be configurable.

**Comment by pkozlowski-opensource** [comment]
I can confirm that this is a bug in Angular v9: https://stackblitz.com/edit/angular-ivy-rzecw9?file=src%2Fapp%2Fapp.component.ts

**Comment by AGreat1** [comment]
Just wanted to mention, that this bug was actually useful for me to solve a problem. I propose this be promoted to a documented feature instead of fixing it by removal.

**Comment by JeanMeche** [comment]
I've opened #50385 to include static in the API !

**Comment by pkozlowski-opensource** [comment]
@JeanMeche I'm actually not sure if we should add this option as it might be confusing.

For the view queries the query author and the query space (template) are authored by the same person. So a developer can decide / know if there are embedded view (if, for etc.) involved or not.

For content queries the query author and content author are 2 _very_ different roles. And what might be happening that people use the `static` option to do "hey, I can get results earlier" but then be confused why not all results are reported.

Honestly, I'm on a fence here and again, I'm afraid that it might result in hard to diagnose errors. At the same time I'm not sure what is the use-case for having static option on content children ("it happens to work" is not enough :-) )

**Comment by JeanMeche** [comment]
Yeah it make sense. This raises another question, what is the rationale for keeping `static` on the `ContentChild` decorator then ?

**Comment by martinboue** [comment]
This issue is still here in v18.2, the documentation and Typescript definiation are **contradictory** : 

### Documentation :
![image](https://github.com/user-attachments/assets/4ed79ad0-9042-4f3b-993a-79eb33f8a3e1)

### Tyescript :
```
@ContentChildren(MyChild, { static: true }) children: QueryList<MyChild>;
```
results in the following error: `'static' does not exist in type '{ descendants?: boolean | undefined; emitDistinctChangesOnly?: boolean | undefined; read?: any; }'`

index.d.ts file:
```
(selector: ProviderToken<unknown> | Function | string, opts?: {
        descendants?: boolean;
        emitDistinctChangesOnly?: boolean;
        read?: any;
    }): any;
    new (selector: ProviderToken<unknown> | Function | string, opts?: {
        descendants?: boolean;
        emitDistinctChangesOnly?: boolean;
        read?: any;
    }): Query;
```

**Either documentation or typescript type needs updating!**

### Workaround
Wait for `ngAfterContentInit` or cast options:
```
@ContentChildren(MyChild, { static: true } as any) children: QueryList<MyChild>;
```
